Automattic/wp-calypso
# My Home: Add Promote Post to my home page swipeable

#### Intro
Hi Team! This is my first PR in Calypso and the first PR of a series of PRs to add entry points in Calypso for a new feature: allow WordPress.com users to pay for advertising to promote their posts on various A8C products. 
For details please see this P2 A self-serve ad client for WPCOM (pbpazC-2Pd-p2).  
The majority of the feature is done outside of WP.com as a standalone widget, here we only need to add buttons to Calypso to show the widget with required parameters.
This PR does **NOT** have all the code for the feature to work. In order to make the PRs smaller, we've put it behind a **feature flag `promote-post`**, and it's hidden for now. So we can iterate quickly.

#### Proposed Changes
This PR adds a swipeable task to My Home page on top, when the promote-post feature is enabled, it will show up and allow user to click on the button to invoke the widget (or jump to another route, the behaviour is not set in stone yet). 

#### Testing Instructions
Since the feature is behind a toggle and hidden. Also the actual widget isn't deployed yet. So here we only need to test that, when the feature is not enabled, it does not show up in the task list. And when it is enabled, it should show up. (No need to test clicking the button, we'll iterate on that in later PRs).

1. Check out the branch.
2. Load up the My Home page of your favourite testing blog, check that the Promote Post is NOT there.
3. Enable the `promote-post` feature, and load the page again. Check now that the Promote Post should show up.

![Screen Shot 2022-07-27 at 10 12 07](https://user-images.githubusercontent.com/6070516/181121225-0c7a31dc-038c-4f8e-9716-e97e3b061d02.jpg)



*

#### Pre-merge Checklist



- [ ] Have you sent any new strings for translation (PCYsg-1vr-p2) ASAP? (We'll do that once the strings are final, I'll create a ticket for that).
